Binary distributions

kleetus opened the pull request.

braydonf commented on README.md:
```diff
@@ -151,7 +151,7 @@ $ cd benchmarks
-## Shared Library Patch
+## Static Library Patch
 
 To provide native bindings to JavaScript *(or any other language for that matter)*, Bitcoin code, itself, must be linkable. Currently, Bitcoin Core provides a JSON RPC interface to bitcoind as well as a shared library for script validation *(and hopefully more)* called libbitcoinconsensus. There is a node module, [node-libbitcoinconsensus](https://github.com/bitpay/node-libbitcoinconsensus), that ...
```
compile as a static library.

braydonf commented on package.json:
```diff
@@ -57,5 +59,13 @@
     "sinon": "^1.15.4"
+  },
+  "bundledDependencies": ["node-pre-gyp"],
+  "binary": {
```
spacing (should be two spaces)

braydonf commented on bin/clean:
```diff
@@ -0,0 +1,7 @@
+root_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)/.."
+cd "${root_dir}"
+
+rm -fr libbitcoind
```
should we remove this directory or run make clean in it?

braydonf commented on binding.gyp:
```diff
@@ -27,14 +27,33 @@
-        '<!(./platform/os.sh filesystem)',
-        '<!(./platform/os.sh thread)',
-        '<!(./platform/os.sh lib)'
+        '<!(./variables.sh bitcoind)',
```
should variables.sh go in the "bin" directory with the build scripts?

kleetus commented:
I've just addressed the concerns from Braydon. I've moved the varaibles.sh script to bin, make clean in libbitcoind, cached only the static lib for libbitcoind.a and fixed the README.md to refer to only the static library.

braydonf commented:
Do we still need the `Makefile` in the base directory? It references node-gyp still.

braydonf commented on README.md:
````diff
@@ -111,7 +111,7 @@ To run tests against the bindings, as defined in `bindings.gyp` the regtest feat
 ```bash
 export BITCORENODE_ENV=test
-rm -rf platform/<os_name>/*
+node-pre-gyp clean
````
Running these commands after a build doesn't work.

braydonf commented on bin/build-libbitcoind:
```diff
@@ -149,38 +137,20 @@ apply the current patch from "${root_dir}"/etc/bitcoin.patch? (y/N): "
+  prepare_for_caching
   echo 'Build finished successfully.'
 else
   echo 'Using existing shared library.'
```
Using existing static library?

braydonf commented on bin/clean:
```diff
@@ -0,0 +1,7 @@
+root_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)/.."
+cd "${root_dir}"
+
+make clean
```
shouldn't this make clean be in the btc_dir?

braydonf commented:
When running `npm start`, there was an error with `undefined symbol: getaddrinfo_a`

braydonf commented on package.json:
```diff
@@ -39,6 +39,7 @@
   "dependencies": {
+    "node-pre-gyp": "0.5.x",
     "async": "1.3.0",
     "bindings": "^1.2.1",
```
We don't need the bindings module anymore.

braydonf commented on README.md:
````diff
@@ -130,7 +130,7 @@ mocha -R spec integration/regtest.js
 
 ```bash
-$ node-gyp -d rebuild
+$ node-pre-gyp -d rebuild
````
Also I think it would be good to document that `node-pre-gyp` can be installed via running `npm install node-pre-gyp -g` if it hasn't already been installed (before the first time it's needed).

braydonf commented on binding.gyp:
```diff
@@ -27,14 +27,33 @@
+    "type": "none",
+    "dependencies": [ "<(module_name)" ],
+    "copies": [
+    {
```
Indentation

braydonf commented:
Went through and tested the build and found a few more issues, and areas that need an update. The regtests passed locally.

braydonf commented:
LGTM

pnagurny commented on RELEASE.md:
```diff
@@ -0,0 +1,34 @@
+
+### How to Release
+
+Ensure you've followed the instructions in the README.md for building the project from source. You will be using the node-gyp to build buildings and a script upload the binary to S3. You will also need credentials for BitPay's bitcore-node S3 bucket and be listed as an author for the bitcore-node's npm module.
```
build buildings?

pnagurny commented:
LGTM. Builds on my machine.
